### PR TITLE
use arguments from form, and use json files uploaded in the form

### PR DESCRIPTION
**Pull request by championswimmer**
Also fixes travis tests and heroku auto-deploy

### CODE REVIEW

**Comment by championswimmer** [comment]
yay!!! travis works

File: src/generator/app.js [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+  {name: 'locationfile', maxCount: 1}
+]);
+
 app.set('port', (process.env.PORT || 5000));
 
 // Use the www folder as static frontend
 app.use('/', express.static(__dirname + '/www'));
 
-app.all('/generate', function(req, res) {
+app.post('/generate', uploadedFiles, function(req, res, next) {
```
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: ['next' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l25)

File: src/generator/app.js [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
+var upload = multer({ dest: 'uploads/' });
+
 var generator = require('./backend/generator.js');
 
+var uploadedFiles = upload.fields([
+  {name: 'speakerfile', maxCount: 1},
+  {name: 'sessionfile', maxCount: 1},
+  {name: 'trackfile', maxCount: 1},
+  {name: 'sponsorfile', maxCount: 1},
+  {name: 'eventfile', maxCount: 1},
```
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l16)

File: src/generator/app.js (hunk 1 above)
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l25)

File: src/generator/app.js [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
 
+var upload = multer({ dest: 'uploads/' });
+
 var generator = require('./backend/generator.js');
 
+var uploadedFiles = upload.fields([
+  {name: 'speakerfile', maxCount: 1},
+  {name: 'sessionfile', maxCount: 1},
+  {name: 'trackfile', maxCount: 1},
+  {name: 'sponsorfile', maxCount: 1},
```
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l15)

File: src/generator/app.js (hunk 1 above)
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l25)

Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l25)

File: src/generator/app.js [hunk 4]
```diff
@@ -1,18 +1,32 @@
 'use strict';
 
 var express = require('express');
+var multer = require('multer');
 var app = express();
 
+var upload = multer({ dest: 'uploads/' });
```
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Combine this with the previous 'var' statement.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l7)

File: src/generator/app.js (hunk 1 above)
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Unexpected function expression.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l25)

File: src/generator/app.js [hunk 5]
```diff
@@ -1,18 +1,32 @@
...
+
 var generator = require('./backend/generator.js');
 
+var uploadedFiles = upload.fields([
```
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Unexpected var, use let or const instead.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l11)

File: src/generator/app.js [hunk 6]
```diff
@@ -1,18 +1,32 @@
...
+  {name: 'speakerfile', maxCount: 1},
+  {name: 'sessionfile', maxCount: 1},
+  {name: 'trackfile', maxCount: 1},
```
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2729341163/issues/source?bid=3233857&fileBranchId=3349103#l14)

**Comment by championswimmer** [comment]
@juslee 
We are using single quotes in strings. Can you make codacy accept that, instead of asking for double quotes ?

**Comment by aayusharora** [comment]
looks good, :+1:

**Pull request merged**